src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BytesKmerizer.java

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * Iterator over successive Kmers from a byte[].
+ * Bytes are interpreted as ASCII characters.
+ * Silently skips over parts of the sequence that has characters other than A, C, G, or T.
+ *
+ * Created by tsharpe on 12/18/15.
+ */
+public class BytesKmerizer implements Iterator<Kmer>
+{
+    private final byte[] seq;
+    private final int kSize;
+    private int idx;
+    private Kmer kmer;
+
+    public BytesKmerizer( final byte[] someSeq, final int someKSize )
+    {
+        seq = someSeq;
+        kSize = someKSize;
+        idx = 0;
+        kmer = new Kmer(kSize);
+        fill(0);
+    }
+
+    @Override
+    public boolean hasNext()
+    {
+        return kmer != null;
+    }
+
+    @Override
+    public Kmer next()
+    {
+        if ( kmer == null ) throw new NoSuchElementException("Kmerization sequence exhausted.");
+        final Kmer result = kmer;
+        fill(kSize -1);
+        return result;
+    }
+
+    public static Kmer toKmer( final byte[] seq )
```

Comment:
I'd argue that if creating a Kmer from a set of bases is a pretty reasonable operation to perform.  Since our standard representation of bases is as a `byte[]`, I don't see it being a problem to bake that into Kmer itself.